go/vt/vtgate/executor.go

```diff
@@ -228,18 +236,19 @@ func (e *Executor) handleExec(ctx context.Context, safeSession *SafeSession, sql
 			normalized := sqlparser.String(stmt)
 			sql = normalized + comments
 		}
-
+		logStats.PlanTime = execStart.Sub(logStats.StartTime)
 		logStats.SQL = sql
 		logStats.BindVariables = bindVars
-
-		execStart := time.Now()
-		logStats.PlanTime = execStart.Sub(logStats.StartTime)
-
+		if keyRange != nil {
+			stmtType := sqlparser.Preview(sql)
+			if stmtType == sqlparser.StmtInsert {
+				return nil, vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "range queries not supported for inserts: %s", safeSession.TargetString)
+			}
+			return e.handleRangeExec(ctx, safeSession, sql, bindVars, target, keyRange, logStats)
```

Comment:
This early return doesn't update the logStats properly which it does in the shard-targeted branch.

Reply:
Good catch. Updating.

Reply:
Actually, I think it is. It's done inside the method. Am I missing one?

Reply:
Oh right. Well given the asymmetry with the shardExec branch this feels like something is missing.

I'd say we should either move the logStats updates into `shardExec` or pull them out of `shardRangeExec` but doing both seems confusing.

Reply:
Done. While doing this, I realized there was a place missing in the validation return. Fixed.